### ISSUE TITLE
v1 "PROXY UNK" fails with Error::Invalid instead of Error::BufferTooShort

### ISSUE THREAD

**Issue by ramiroaisen** [tibordp/proxy-header#1]
A valid incomplete header containing "PROXY UNK" fails to parse with Error::Invalid when it should fail with Error::BufferTooShort.

"PROXY UNK" is a valid incomplete header, the start of "PROXY UNKOWN\r\n"

See this test I'm running:

```rust
  #[test]
  pub fn issue_repr() {
    use proxy_header::ProxyHeader;

    let buf = b"PROXY UNK"; // NOWN\r\n;
    let config = proxy_header::ParseConfig {
      allow_v1: true,
      allow_v2: false,
      include_tlvs: false,
    };

    let err = ProxyHeader::parse(buf, config)
      .unwrap_err();

    // this fails with Error::Invalid instead of Error::BufferTooShort
    assert!(
      matches!(err, proxy_header::Error::BufferTooShort),
    )
  }
```

**Comment by tibordp** [comment]
Hi, thank you for reporting! 

I cannot repro with your specific example, but the issue is indeed there for `PROXY UNKNOW`

I will have the fix up shortly.

**Comment by tibordp** [comment]
Hi, fixed version 0.1.2 has been uploaded to crates.io

**Comment by ramiroaisen** [comment]
> Hi, fixed version 0.1.2 has been uploaded to crates.io

Wow, that was fast, Thank you very much!
